app/controllers/concerns/localized.rb

```diff
@@ -17,12 +17,12 @@ def set_locale
   end
 
   def default_locale
-    ENV.fetch('DEFAULT_LOCALE') do
-      user_supplied_locale || I18n.default_locale
-    end
-  end
+    env = ENV['DEFAULT_LOCALE']
 
-  def user_supplied_locale
-    http_accept_language.language_region_compatible_from(I18n.available_locales)
+    http_accept_language.preferred_language_from([env]) ||
```

Comment:
This is a lot of conditionals happening in one method...

What about breaking this out to something like...

`request_locale || env_locale || I18n.default_locale`

The impl of `request_locale` would then be like `preferred_locale || compatible_locale` and those methods would then check both the env and available.